src/main.rs

```diff
@@ -305,15 +313,70 @@ fn main() -> Result<()> {
         file_list.extend(u);
     }
 
+    let mut all_files = Vec::<String>::new();
+
     if let Some(file_names) = matches.get_many::<String>("INPUT") {
-        file_list.push(FileBundle {
-            include_dirs: include_dirs.clone(),
-            export_incdirs: HashMap::new(),
-            defines: defines.clone(),
-            files: file_names.map(String::from).collect(),
+        all_files.extend(file_names.map(|x| x.to_string()).collect::<Vec<_>>());
+    }
+
+    for path in matches.get_many::<String>("flist").into_iter().flatten() {
+        let file = File::open(path).unwrap_or_else(|e| {
+            eprintln!("error opening `{}`: {}", path, e);
+            process::exit(1)
         });
+        let lines = BufReader::new(file).lines();
+
+        let proper_lines: Vec<String> = lines.filter_map(|x| x.ok()).collect();
+
+        all_files.extend(proper_lines);
     }
 
+    let mut stdin_incdirs = include_dirs;
+    let mut stdin_defines = HashMap::<String, Option<String>>::new();
+
+    let stdin_files = all_files
+        .into_iter()
+        .map(String::from)
+        .filter_map(|file_str| {
+            let split_str = file_str.splitn(3, '+').collect::<Vec<_>>();
```

Comment:
So theoretical question here: we only support filename+define=asdf+incdir=adfasdf and not multiple defines. I mean, one can probably chain them with `,`. Maybe we could add an error if a + appears in the last split: `split_str[2]`?

Reply:
So, to differentiate the options a space in between is required. This means `/path/to/file` gives a file, `+define+KEY` or `+define+KEY=VALUE` should add a define, while `+incdir+/include/directory` should add another include directory. There can be multiple of these, with spaces (or newlines in the flist) between.

Reply:
Basically anything that has a `define` or `incdir` in between the first and second occurrence of a `+` symbol is interpreted as a define or incdir, otherwise the string is interpreted as a file. I don't think a further error needs to be specified, and any other `+` in the string should not have an impact. For any file path containing two `+` symbols an error is already printed (unless it contains exactly `incdir` or `define` in between), but in a normal case you may want to have a `+` in the last split (e.g. a fancy define statement). The only other check that may be relevant is to ensure that nothing is before the first `+` (i.e. `split_str[0].len() == 0`).